pyvista/core/filters/data_set.py

```diff
@@ -193,6 +200,248 @@ def align(
+        we set the :attr:`~pyvista.Prop3D.user_matrix` as the inverse.
+
+        >>> axes_local = pv.AxesAssembly(
+        ...     scale=aligned.length,
+        ...     user_matrix=inverse,
+        ...     labels=["X'", "Y'", "Z'"],
+        ... )
+
+        Plot the original mesh with its local axes, along with the algned mesh and its
+        axes.
+
+        >>> axes_aligned = pv.AxesAssembly(scale=aligned.length)
+        >>> pl = pv.Plotter()
+        >>> # Add aligned mesh with axes
+        >>> _ = pl.add_mesh(aligned)
+        >>> _ = pl.add_actor(axes_aligned)
+        >>> # Add original mesh with axes
+        >>> _ = pl.add_mesh(
+        ...     mesh, style='wireframe', color='black', line_width=3
+        ... )
+        >>> _ = pl.add_actor(axes_local)
+        >>> pl.show()
+        """
+
+        def _validate_vector(vector, name):
+            if vector is not None:
+                if isinstance(vector, str):
+                    vector = vector.lower()
+                    valid_strings = list(NORMALS.keys())
+                    _validation.check_contains(item=vector, container=valid_strings, name=name)
+                    vector = NORMALS[vector]
+                vector = _validation.validate_array3(vector, dtype_out=float, name=name)
+            return vector
+
+        axes, std = pyvista.principal_axes(self.points, return_std=True)
+
+        # Set directions to +X,+Y,+Z by default
+        if axis_0_direction is None and axis_1_direction is None and axis_2_direction is None:
+            axis_0_direction = (1.0, 0.0, 0.0)
+            axis_1_direction = (0.0, 1.0, 0.0)
+            axis_2_direction = (0.0, 0.0, 1.0)
+        else:
+            axis_0_direction = _validate_vector(axis_0_direction, name='axis 0 direction')
+            axis_1_direction = _validate_vector(axis_1_direction, name='axis 1 direction')
+            axis_2_direction = _validate_vector(axis_2_direction, name='axis 2 direction')
+
+        # Swap any axes which have equal std (e.g. so that we XYZ order instead of YXZ order)
+        # Note: Swapping may create a left-handed coordinate frame. This is fixed later.
+        axes = _swap_axes(axes, std)
+
+        # Maybe flip directions of first two axes
+        if axis_0_direction is not None and np.dot(axes[0], axis_0_direction) < 0:
+            axes[0] *= -1
+        if axis_1_direction is not None and np.dot(axes[1], axis_1_direction) < 0:
+            axes[1] *= -1
+
+        # Ensure axes form a right-handed coordinate frame
+        if np.linalg.det(axes) < 0:
+            axes[2] *= -1
+
+        # Maybe flip direction of third axis
+        if axis_2_direction is not None:
+            has_correct_sign = np.dot(axes[2], axis_2_direction) >= 0
+            other_directions_are_specified = (
+                axis_0_direction is not None and axis_1_direction is not None
+            )
+            if has_correct_sign or other_directions_are_specified:
+                pass
```

Comment:
Is this the change you intend?
```suggestion
            if not (has_correct_sign or other_directions_are_specified):
                raise ValueError("message of error.")
```

Reply:
Should be fixed with https://github.com/pyvista/pyvista/pull/6485/commits/9cc91aadfed05b01a44a3fde069a62168a2926ba